ballot/models.py

```diff
@@ -52,6 +52,10 @@ def __str__(self):
 class BallotItemResponse(models.Model):
     """
     YES/NO to a referendum, or a candidate.
+
+    This is a *conceptually* abstract class, but exists
+    as a Django model so that finance can point to propositions
+    and candidates equally.
     """
```

Comment:
So, just talking about referenda, there will be two BallotItemResponse’s for each BallotItem, right?  One in favor and one opposed.  I assume that is so we can associate spending for and against the measure with each “Response”, right?  So why does the class have the fields it has? Particularly both the “pro” and the “con” statements?  I would think if an instance of this class was for the “YES” it would have the pro statement and if it was the “NO” it would have the con statement, maybe.  Really I think all the fields currently in this class aught to be somewhere else, maybe in the BallotItem itself or in a BallotItemDescription abstract class which can be subclassed to a BallotMeasureDescription and a BallotOfficeDescription.  Although the latter may not be needed.

BallotItemResponce would then must have a field that says “For” or “Against” for referenda and, I suppose, a reference to a candidate if its an office.

I think the name BallotItemChoice is better than BallotItemResponce as the ballot contains choices, not responses.  

> On Jan 14, 2016, at 11:32 AM, Ben Cipollini <notifications@github.com> wrote:
> 
> In ballot/models.py <https://github.com/caciviclab/disclosure-backend/pull/134#discussion_r49773167>:
> 
> > @@ -52,6 +52,10 @@ def __str__(self):
> >  class BallotItemResponse(models.Model):
> >      """
> >      YES/NO to a referendum, or a candidate.
> > +
> > +    This is a *conceptually* abstract class, but exists
> > +    as a Django model so that finance can point to propositions
> > +    and candidates equally.
> >      """
> You're not a pill at all, this is wonderful feedback. I really appreciate it, and appreciate your patience to walk through it with me--I also feel I'm being a pill :)
> 
> This is subclassed for candidates here:
> https://github.com/caciviclab/disclosure-backend/blob/master/office_election/models.py#L101 <https://github.com/caciviclab/disclosure-backend/blob/master/office_election/models.py#L101>
> So, just to be clear, it's definitely not a response from each person. It's a potential response--a bubble you could fill on the ballot, an answer to a specific ballot item question.
> 
> BallotItemDescription sounds like a description of the entire ballot item. This is not that. There is one BallotItemResponse per possible responses to the ballot item, and each has information related to that response (title: yes/no/caniddate name, information in favor / opposing that particular response). So for each BallotItem, you need multiple BallotItemResponse to fully describe the entire item on the ballot.
> 
> As far as fields on the BallotItemResponse object, I agree that it's very much structured towards a referendum. I think it can be abstracted better. Lemme think about use-cases, but I think moving most of the properties onto Candidate and ReferendumResponse is a good idea.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/caciviclab/disclosure-backend/pull/134/files#r49773167>.
>